File: soap-client/src/main/java/ru/romanow/protocols/soap/web/DocumentWrappedWebServiceClient.java
package ru.romanow.protocols.soap.web;

import org.apache.commons.lang3.RandomStringUtils;
import org.apache.commons.lang3.RandomUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import ru.romanow.protocols.soap.generated.wrapped.model.TestObjectRequest;
import ru.romanow.protocols.soap.generated.wrapped.model.TestObjectResponse;
import ru.romanow.protocols.soap.generated.wrapped.model.WebServiceDocumentWrapped;

public class DocumentWrappedWebServiceClient
        implements WebServiceClient {
    private static final Logger logger = LoggerFactory.getLogger(WebServiceClient.class);

    private final WebServiceDocumentWrapped webService;

    public DocumentWrappedWebServiceClient(WebServiceDocumentWrapped webService) {
        this.webService = webService;
    }

    @Override
    @SuppressWarnings("Duplicates")
    public void makeRequest() {
        TestObjectRequest request = new TestObjectRequest();
        request.setId(RandomUtils.nextInt(0, 1000));
        request.setSearchString(RandomStringUtils.randomAlphabetic(10));

        logger.info("Request [{}]", request);

        TestObjectResponse response = webService.processRequest(request);
        logger.info("Response [{}]", response);
    }
}

File: soap-client/src/main/java/ru/romanow/protocols/soap/web/DocumentWrappedWebServiceClient.kt
package ru.romanow.protocols.soap.web

import com.google.common.base.MoreObjects
import org.apache.commons.lang3.RandomStringUtils
import org.apache.commons.lang3.RandomUtils
import org.slf4j.LoggerFactory
import ru.romanow.protocols.soap.generated.wrapped.model.TestObjectRequest
import ru.romanow.protocols.soap.generated.wrapped.model.WebServiceDocumentWrapped

class DocumentWrappedWebServiceClient(
    private val webService: WebServiceDocumentWrapped
) : WebServiceClient {
    private val logger = LoggerFactory.getLogger(WebServiceClient::class.java)

    override fun makeRequest() {
        val request = TestObjectRequest()
        request.id = RandomUtils.nextInt(0, 1000)
        request.searchString = RandomStringUtils.randomAlphabetic(10)
        logger.info(
            "Request [{}]", MoreObjects.toStringHelper(request)
                .add("id", request.id)
                .add("searchString", request.searchString)
                .toString()
        )
        val response = webService.processRequest(request)
        logger.info(
            "Response [{}]", MoreObjects.toStringHelper(response)
                .add("code", response.code)
                .add("data", response.data)
                .toString()
        )
    }
}